mhw_library/measures.py

```diff
@@ -0,0 +1,605 @@
+            coords = list(zip(lats, long_range(lons)))
+
+            for ind in range(len(coords)-1):
+                distances.append(haversine(coords[ind], coords[ind+1],Unit.KILOMETERS))
+            
+            distances.append(haversine(coords[-1], coords[0],Unit.KILOMETERS))
+        
+        perimeters.append(np.sum(distances))
+    
+    return perimeters
+
+def calc_perimeter_vs_area(
+    spatial_extents : list, 
+    perimeters : list) -> np.ndarray:
+    """
+    Calculates the perimeter versus area of an object
+    at each timestamp 
+    
+    Parameters
+    ----------
+    spatial_extents : list of areas
+    perimeters : list of perimeters
+    
+    Returns
+    ----------
+    np.ndarray of percentages of perimeter versus spatial extent
+    
+    Notes
+    ----------
+    Gives an idea of how deformed an object is.
+    Also gives an idea of optimal shapes.
+    We have squared units so that we return a unitless number.
+    """
+    return ((np.asarray(perimeters)**2)/np.asarray(spatial_extents))
+
+def calc_complement_to_deformormation(coords_full: list, spatial_extents: list) -> np.ndarray:
+    """
+    Calculates the complement to the deformation, which is the fraction of overlapped domain
+    between consecutive timestamps.
+    
+    Parameters
+    ----------
+    coords_full : list of arrays
+        A list of numpy arrays where each array contains the latitude and
+        longitude coordinates where the object has an imprint at each timestamp.
+    spatial_extents : list of areas
+        A list of spatial extents where each element corresponds to the area
+        covered by the object at each timestamp.
+    
+    Returns
+    -------
+    np.ndarray
+        An array of the ratio of shared area between consecutive timestamps.
+    
+    Notes
+    -----
+    This function is my implementation of a measure from Sun, D., Jing, Z., Li, F., 
+    Wu, L., "Characterizing Global Marine Heatwaves Under a Spatio-temporal Framework," 
+    Progress in Oceanography (2022).
+    """
+    sharedarea_ls = []
+    
+    for i in range(len(coords_full)-1):
+        a_set = set(coords_full[i])
+        b_set = set(coords_full[i+1])
+        
+        if a_set & b_set:
+            coords = a_set & b_set
+            y,x=zip(*coords)
+            dlon = [np.cos(y[c]*np.pi/180)*(111.320*1) for c in np.arange(0, len(coords))]
+            dlat = (110.574 *1) * np.ones(len(dlon))
+            sharedarea = np.sum(dlon*dlat)
+            sharedarea_ls.append((sharedarea/ (spatial_extents[i] + spatial_extents[i+1])))
+        
+        else:
+            sharedarea_ls.append(0)
+            
+    return np.array(sharedarea_ls)
+
+def calc_deformation(sharedarea_ls: list) -> np.ndarray:
+    """
+    Calculates the deformation, which is a measure of the extent to which an 
+    object deviates from a smooth, circular shape. The deformation is defined 
+    as the fraction of non-overlapped domain occupied by the object at 2 
```

Comment:
non-overlapped over union

Reply:
Calculates the deformation, which is a measure of the extent to which an object deviates(in terms of mismatch area) from its original shape.